tests/functional/tokenstorage/test_simplejson_file.py

```diff
@@ -10,44 +10,44 @@
 
 
 @pytest.fixture
-def filename(tempdir):
-    return os.path.join(tempdir, "mydata.json")
+def json_file(tmp_path):
+    return tmp_path / "mydata.json"
 
 
-def test_file_does_not_exist(filename):
-    adapter = SimpleJSONFileAdapter(filename)
+def test_file_does_not_exist(json_file):
+    adapter = SimpleJSONFileAdapter(json_file)
     assert not adapter.file_exists()
 
 
-def test_file_exists(filename):
-    open(filename, "w").close()  # open and close to touch
-    adapter = SimpleJSONFileAdapter(filename)
+def test_file_exists(json_file):
+    open(json_file, "w").close()  # open and close to touch
+    adapter = SimpleJSONFileAdapter(json_file)
     assert adapter.file_exists()
 
 
-def test_store(filename, mock_response):
-    adapter = SimpleJSONFileAdapter(filename)
+def test_store(json_file, mock_response):
+    adapter = SimpleJSONFileAdapter(json_file)
     assert not adapter.file_exists()
     adapter.store(mock_response)
 
-    with open(filename) as f:
+    with open(json_file) as f:
         data = json.load(f)
```

Comment:
```suggestion
    data = json.loads(json_file.read_text())
```

Reply:
Good call! I didn't look that carefully for opportunities to switch to the pathlib APIs, but we should take them all, where we can.